sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ArrayBasedMapBuilder.scala

```diff
@@ -47,13 +48,17 @@ class ArrayBasedMapBuilder(keyType: DataType, valueType: DataType) extends Seria
   private lazy val keyGetter = InternalRow.getAccessor(keyType)
   private lazy val valueGetter = InternalRow.getAccessor(valueType)
 
-  def put(key: Any, value: Any): Unit = {
+  def put(key: Any, value: Any, withSizeCheck: Boolean = false): Unit = {
     if (key == null) {
       throw new RuntimeException("Cannot use null as map key.")
     }
 
     val index = keyToIndex.getOrDefault(key, -1)
     if (index == -1) {
+      if (withSizeCheck && size >= ByteArrayMethods.MAX_ROUNDED_ARRAY_LENGTH) {
```

Comment:
I think we should aways check the size. Such a big map is very likely to cause problems.

Reply:
this flag is just for perf reasons, we can skip the check in some conditions and I didn't want to introduce perf overhead if not needed. If we remove the flag we would do the comparison for each item, also when it is not needed.

Reply:
hmmm, I'd like to avoid premature optimization. Actually how much perf this can save? This code block is already doing some heavy work.

Reply:
ok, let me remove it then, thanks.